File: chapter11/typescript/LinuxOS.ts
import { OperatingSystem } from "./OperatingSystem";

/**
 * LinuxOS implements the bridge "implementation" interface OS
 *
 * @Author Bridget
 */
 export class LinuxOS implements OperatingSystem {

    public displayOSName(): void {
        console.log("Ubuntu");
    }

    public displayOSVersion(): void {
        console.log("20.04 LTS");
    }

    public displayMemory(): void {
        console.log("128 GB RAM");
    }

    public start(): void {
        console.log("Start Linux Operating System");
    }

    public shutdown(): void {
        console.log("Shutdown Linux Operating System");
    }
}

File: chapter11/java/designpattern/src/main/java/com/designpattern/LinuxOS.java
package com.designpattern;

/**
 * LinuxOS implements the bridge "Implementor" interface OperatingSystem
 *
 * @version 1.0.0
 * @Author Bridget
 * @license MIT
 */
public class LinuxOS implements OperatingSystem {

    @Override
    public void displayOSName() {
        System.out.println("Ubuntu");
    }

    @Override
    public void displayOSVersion() {
        System.out.println("20.04 LTS");
    }

    @Override
    public void displayMemory() {
        System.out.println("128 GB RAM");
    }

    @Override
    public void start() {
        System.out.println("Start Linux Operating System");
    }

    @Override
    public void shutdown() {
        System.out.println("Shutdown Linux Operating System");
    }
}
